test/v1/models/request/CreateAndAmendCharitableGivingTaxRelief/GiftsSpec.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.request.CreateAndAmendCharitableGivingTaxRelief
+
+import play.api.libs.json.{JsValue, Json}
+import support.UnitSpec
+import v1.models.request.createAndAmendCharitableGivingTaxRelief.{Gifts, NonUkCharities}
+
+class GiftsSpec extends UnitSpec {
+
+  val nonUkCharitiesModel: NonUkCharities =
+    NonUkCharities(
+      charityNames = Some(Seq("abcdefghijklmnopqr")),
+      totalAmount = 10000.89
+    )
+
+  val model: Gifts =
+    Gifts(
+      nonUkCharities = Some(nonUkCharitiesModel),
+      landAndBuildings = Some(10000.89),
+      sharesOrSecurities = Some(10000.89),
+    )
+
+  val mtdJson: JsValue = Json.parse(
+    """
+      |{
+      |   "nonUkCharities": {
+      |      "charityNames":[
+      |         "abcdefghijklmnopqr"
+      |      ],
+      |      "totalAmount": 10000.89
+      |   },
+      |   "landAndBuildings":10000.89,
+      |   "sharesOrSecurities":10000.89
+      |}
+      |""".stripMargin)
+
+  val desJson: JsValue = Json.parse(
+    """
+      |{
+      |   "investmentsNonUkCharitiesCharityNames":[
+      |      "abcdefghijklmnopqr"
+      |   ],
+      |   "investmentsNonUkCharities":10000.89,
+      |   "landAndBuildings":10000.89,
+      |   "sharesOrSecurities":10000.89
+      |}
+      |""".stripMargin)
+
+  "reads" when {
+    "passed valid JSON" should {
+      "return a valid model" in {
+        model shouldBe mtdJson.as[Gifts]
```

Comment:
Please switch around:
```suggestion
        mtdJson.as[Gifts] shouldBe model
```